Adaboost

parismita opened the pull request.

aldro61 commented:
I'm not sure why, but there are some problems with the c++ solver on Mac. It works perfectly on linux. I'll take a look.

aldro61 commented:
It's fixed! I'm having a hard time pushing the changes, but I'll do it this weekend.

parismita commented:
@aldro61 the mmitboost and mmitboost.predict has most of the code same...as we cant find the prediction value without running it, and as its iterative process hence mmitboost cant be used for prediction

So is there any way to improve this?

aldro61 commented on Rpackage/R/mmitboost.R:
```diff
@@ -0,0 +1,83 @@
+    weights <- weights*exp(alpha*scores)
+    final_scores <- final_scores + alpha*prediction
+  }
+  #plot(tree)
```
Please remove commented lines.

aldro61 commented:
There might be some code that can be reused, maybe packaged into one or many utility functions. But before that, we should make sure that the implementation is ok.

I think that the Adaboost algorithm that you implemented is for classification. The algorithm is different for regression. Scikit-learn implements the following version (Adaboost.R2) https://www.researchgate.net/publication/2424244_Improving_Regressors_Using_Boosting_Techniques.

Note that according to this algorithm, the training data needs to be resampled with replacement before training each tree. Also, there seems to be a learning rate hyperparameter that can be tuned by the user.

parismita commented:
ok....i'll check that and update

Commit:
Merge pull request #9 from parismita/gsoc_doc

Gsoc doc

Commit:
Revert "Gsoc doc"

Commit:
Merge pull request #10 from parismita/revert-9-gsoc_doc

Revert "Gsoc doc"

parismita committed:
basic implementation of adaboost for regression

parismita committed:
adaboost algorithm implemented for regression
return result including trees, B value, predictions of each tree

parismita committed:
update mmit.predict

parismita committed:
add example for dev purpose

parismita committed:
solve conflict

Merge branch 'master' of https://github.com/aldro61/mmit

Conflicts:
	Rpackage/R/mmif.R
	Rpackage/R/mmif.cv.R
	Rpackage/R/mmit.cv.R
	Rpackage/R/mmitboost.R
	Rpackage/R/mmitboost.predict.R
	Rpackage/man/mmit.Rd
	Rpackage/src/registerDynamicSymbol.c

parismita committed:
fix cost bugs mmitboost

Commit:
Merge branch 'master' into master

parismita commented on Rpackage/R/mmitboost.R:
```diff
@@ -0,0 +1,87 @@
+  upper = prediction - target.mat[,2] + margin
+  upper[upper < 0] <- 0
+  if(loss == 'hinge'){
+    cost = lower + upper
```
@aldro61 How to make sure that cost is between 0 to 1?

parismita committed:
weigthed mean prediction

parismita committed:
Merge branch 'master' of https://github.com/parismita/mmit

parismita commented on Rpackage/R/mmitboost.predict.R:
```diff
@@ -0,0 +1,33 @@
+#' 
+#' @export
+mmitboost.predict <- structure(function(target.mat, mmitboost_results) {
+  final_scores <- weighted.mean(mmitboost_results$pred, mmitboost_results$B)
```
@aldro61  How to implement weighted median?

![Selection_083](https://user-images.githubusercontent.com/17319560/61592278-d3790280-abee-11e9-9736-e627843a7a22.png)

Commit:
Merge branch 'master' into master

Commit:
Rename tutorial.Rmd to README.Rmd

Commit:
Rename README.Rmd to README.md

Commit:
Rename README.md to tutorial.Rmd
